### PR TITLE
Fixes #13, Update Server Configuration

### LINKED ISSUE #13: Docker Container Initialization Error with louis-db
**Issue Description:**  
Error when trying to launch postgres.sh in louis-db-server.

**Steps to Reproduce:**  
1. Clone the `louis-db` repository.
2. Go to the branch `issue8-missing-chunks` to have access to the bin folder
3. Create a symbolic link to `~/src/louis-db/bin/postgres.sh` using the command: `ln -s ~/src/louis-db/bin/postgres.sh`.
4. Execute `./postgres.sh`.
5. Check the active Docker containers using `docker ps`.
6. Observe that the expected Docker container for louis-db is missing.
7. To investigate further, check the logs of the louis-db server with the command: `docker logs louis-db-server`.

**Error Message:**
```
chmod: /var/run/postgresql: Operation not permitted
Error: Database is uninitialized and superuser password is not specified.
       You must specify POSTGRES_PASSWORD to a non-empty value for the
       superuser. For example, "-e POSTGRES_PASSWORD=password" on "docker run".

       You may also use "POSTGRES_HOST_AUTH_METHOD=trust" to allow all
       connections without a password. This is *not* recommended.

       See PostgreSQL documentation about "trust":
       https://www.postgresql.org/docs/current/auth-trust.html
```
**Expected Behavior:**
The Docker container for louis-db should be created and initialized without errors.

**Operating System: Windows 10**
- Windows Subsystem for Linux (WSL) Version: 2
- WSL Distribution: Ubuntu
- Visual Studio Version: 1.82.0
- Docker Version: 24.0.5

### PR DESCRIPTION
**Pull request by JolanThomassin**
Fixes #13,
- adding "echo" to specify if the server already exist in postgres.sh.
- commenting code in lib.sh
- slight modification on the createdb line in setup-db-docker.sh

### CODE REVIEW

File: bin/lib.sh [hunk 1]
```diff
@@ -9,13 +9,13 @@ else
...
-        exit 1
-    fi
-done
+# REQUIRED_ENVIRONMENT_VARIABLES="LOUIS_DSN PGBASE PGUSER PGPASSWORD PGHOST OPENAI_API_KEY AZURE_OPENAI_SERVICE LOUIS_SCHEMA"
+# for VARIABLE in $REQUIRED_ENVIRONMENT_VARIABLES; do
+#     if [ -z "${!VARIABLE}" ]; then
+#         echo "Environment variable $VARIABLE is not set"
+#         exit 1
+#    fi
+# done
```
Comment (rngadam) [inline bin/lib.sh:18]: turn this into a function callable by other scripts that want to check prereqs variables outside of setup.

File: bin/postgres.sh [hunk 2]
```diff
@@ -12,8 +12,11 @@ fi
 STATUS=`docker inspect louis-db-server -f '{{.State.Status}}'`
 
 if [ "$STATUS" = "exited" ]; then
+    echo container exist but as exited, restarting
```
Comment (rngadam) [inline bin/postgres.sh:15]: echo "container exist $DB_SERVER_CONTAINER_NAME but has exited, restarting"

File: bin/postgres.sh [hunk 3]
```diff
@@ -12,8 +12,11 @@ fi
...
     docker start louis-db-server
 elif [ "$STATUS" != "running" ]; then
+    # check PGPASWORD before using it
+    echo container does not exist, creating
```
Comment (rngadam) [inline bin/postgres.sh:19]: update here today

File: bin/setup-db-docker.sh [hunk 4]
```diff
@@ -6,7 +6,7 @@ if [ -z "$PGBASE" ]; then
     exit 1
 fi
 DOCKER_EXEC="docker exec -it louis-db-server"
-$DOCKER_EXEC createdb -E utf-8 $PGBASE
+$DOCKER_EXEC createdb -E utf-8 -U postgres $PGBASE
```
Comment (rngadam) [inline bin/setup-db-docker.sh:9]: Use PGUSER? can we swap for PGUSER?

File: bin/setup-db-docker.sh [hunk 5]
```diff
@@ -6,7 +6,7 @@ if [ -z "$PGBASE" ]; then
     exit 1
 fi
 DOCKER_EXEC="docker exec -it louis-db-server"
-$DOCKER_EXEC createdb -E utf-8 $PGBASE
+$DOCKER_EXEC createdb -E utf-8 -U postgres $PGBASE 
 $DOCKER_EXEC $PSQL_ADMIN -c "CREATE USER $USER; ALTER USER $USER WITH SUPERUSER;"
```
Comment (rngadam) [inline bin/setup-db-docker.sh:10]: $USER = ton user (thomassinj)

**Review by rngadam — CHANGES_REQUESTED** [review]

**Comment by JolanThomassin** [comment]
I've introduced a **verification function** in the last commit ([74d69c4](https://github.com/ai-cfia/louis-db/pull/14/commits/74d69c4b8d5bd3826842edd2c787c127e4e4ea64)) called `is_parameter_empty` in `lib.sh` to **validate parameters before their usage**. Currently, this function is implemented specifically for `PGPASSWORD` and `DB_SERVER_CONTAINER_NAME`.

**Comment by JolanThomassin** [comment]
I've applied modification on the **verification function** in the last commit ([0ce5846](https://github.com/ai-cfia/louis-db/pull/14/commits/0ce58462b6e69a8eb78f5d735c27c909b02fb48f)), now you can utilize the function to **validate all environment variables before their usage**, if any of them are undefined or empty, **the execution will stop**.

**Review by rngadam — CHANGES_REQUESTED** [review]

File: bin/lib.sh [hunk 6]
```diff
@@ -9,13 +9,19 @@ else
...
-    if [ -z "${!VARIABLE}" ]; then
-        echo "Environment variable $VARIABLE is not set"
-        exit 1
+check_environment_variables_defined () {
+  variable_not_set=0
+  for VARIABLE in "$@"; do
+      if [ -z "${!VARIABLE}" ]; then
+          echo "Environment variable $VARIABLE is not set"
+          variable_not_set=1
     fi
```
Comment (rngadam) [inline bin/lib.sh:18]: tabulation? please add a bash shell script linter

File: bin/postgres.sh (hunk 3 above)
Comment (rngadam) [reply bin/postgres.sh:21]: more details in your string, including container name

File: bin/postgres.sh [hunk 7]
```diff
@@ -11,9 +11,14 @@ fi
...
 
+check_environment_variables_defined DB_SERVER_CONTAINER_NAME PGPASSWORD 
+
 if [ "$STATUS" = "exited" ]; then
+    echo "container $DB_SERVER_CONTAINER_NAME exist but has exited, restarting"
     docker start louis-db-server
+
 elif [ "$STATUS" != "running" ]; then
+    echo container does not exist, creating
     docker run --name louis-db-server \
```
Comment (rngadam) [inline bin/postgres.sh:22]: this should be using your environment variable...

File: bin/setup-db-docker.sh [hunk 8]
```diff
@@ -6,7 +6,7 @@ if [ -z "$PGBASE" ]; then
     exit 1
 fi
 DOCKER_EXEC="docker exec -it louis-db-server"
```
Comment (rngadam) [inline bin/setup-db-docker.sh:8]: idem

**Review by rngadam — CHANGES_REQUESTED** [review]
also add usage information to the README.md in bin/

File: bin/lib.sh [hunk 9]
```diff
@@ -1,26 +1,28 @@
...
+    variable_not_set=0
+    for VARIABLE in "$@"; do
+        if [ -z "${!VARIABLE}" ]; then
+            echo "Environment variable $VARIABLE is not set"
+            variable_not_set=1
+        fi
+    done
+
+    if [ $variable_not_set -eq 1 ]; then
+        exit 1
```
Comment (rngadam) [inline bin/lib.sh:24]: be nice and tell the user why you are exiting

**Review by rngadam — APPROVED** [review]

**Comment by JolanThomassin** [comment]
In this commit, I have replace existing **variable names with environment variables**. This update **enhances the flexibility** by allowing configuration settings to be **managed externally** through environment variables, making it easier to **adapt to different deployment environments**.

**Review by rngadam — APPROVED** [review]

**Comment by JolanThomassin** [comment]
We observed that the code exhibited a **high degree of specificity towards `louis-db`.** To enhance flexibility, we have initiated a new issue : https://github.com/ai-cfia/louis-db/issues/15

**Pull request merged**